lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_arm.cpp

```diff
@@ -737,12 +744,17 @@ Status NativeRegisterContextLinux_arm::ReadHardwareDebugInfo() {
   m_refresh_hwdebug_info = false;
 
   return error;
+#else  // __aarch64__
+  return arm64::ReadHardwareDebugInfo(m_thread.GetID(), m_max_hwp_supported,
+                                      m_max_hbp_supported);
+#endif // __arm__
```

Comment:
These final #endif we tend to write the `if` as well like `ifdef __arm__`.

Reply:
If it's something very complex then yeah we simplify it because you should just look for the original `if` and read that.

Reply:
It's not automatically enforced so sometimes things are inconsistent yes. This is new code though so it's fine to be more precise.